src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/H2Translator.java

```diff
@@ -355,4 +355,20 @@ public String translateTrue() {
     public String translateFalse() {
         return "FALSE";
     }
+
+    /**
+     * In H2 2.x, checking a boolean field against a boolean represented as a string causes an error.
+     * This code checks if the string is a boolean and processes it as a boolean, not as a string.
+     */
+    @Override
+    public String translate(final K k) {
+       if (k.getConstant() instanceof String) {
+            String s = (String) k.getConstant();
+            if (s.equalsIgnoreCase("TRUE") || s.equalsIgnoreCase("FALSE")) {
```

Comment:
Question: Does `"1"` and `"0"` also represent boolean values?

Reply:
Makes sense. I was confused, because Oracle uses `1` and `0` to represent Booleans.